### ISSUE TITLE
No support for multiple read-only processes

### ISSUE THREAD

**Issue by mattbab** [chriseldredge/Lucene.Net.Linq#33]
My organization is playing with the idea of utilizing this great library to implement a NoSQL-ish data store based on Lucene.NET to improve accessibility from our .NET products.

In my POC testing I've noticed that when multiple processes attempt to open a single index to search, it results in a `LockObtainFailedException` exception when used with (at least) the FSDirectory.

This exception is the result of attempting to blindly open an IndexWriter when instantiating the `LuceneDataProvider` class (LuceneDataProvider.cs:90) after another application (or possibly even the same one) already has an open `LuceneDataProvider` instance.

I've implemented a hack to workaround this limitation for the time being, but wanted to notify you of the issue before losing track of it. I'll describe the hack here for anyone else that encounters it, but am happy to submit a pull request if you feel it is appropriate. 

Thanks.

LuceneDataProvider:107:

``` c#
            get
            {
+                if (Lucene.Net.Index.IndexWriter.IsLocked(directory))
+                    return null;

                lock (sync)
```

LuceneDataProvider:201:

``` c#
            if (writerIsExternal) return;

+            if (IndexWriter != null)
                IndexWriter.Dispose();
        }
```

**Comment by chriseldredge** [comment]
Thanks for reporting this. Not a lot of thought has gone into multiple reader or multiple writer scenarios yet. 2d649d265564a00130c93f59aff465e9d78a06b4 adds your null check in Dispose and adds ReadOnlyLuceneDataProvider for clients that should not create writers and lock the index.

**Comment by chriseldredge** [comment]
Released in v3.2.52 and published on NuGet.org.
